Description: Make get_secret_keys and get_public_keys methods work with no arguments.
Origin: vendor
Bug: https://rt.cpan.org/Public/Bug/Display.html?id=62177
Forwarded: yes
Author: intrigeri <intrigeri+debian@boum.org>
Last-Update: Sun, 16 Oct 2011 00:46:51 +0200

---
 lib/GnuPG/Interface.pm |   30 ++++++++++++++++++------------
 1 files changed, 18 insertions(+), 12 deletions(-)

--- a/lib/GnuPG/Interface.pm
+++ b/lib/GnuPG/Interface.pm
@@ -435,10 +435,12 @@
             # --fixed-list-mode uses epoch time for creation and expiration date strings.
             # For backward compatibility, we convert them back using GMT;
             my $expiration_date_string;
-            if ($expiration_date eq '') {
-              $expiration_date = undef;
-            } else {
-              $expiration_date_string = $self->_downrez_date($expiration_date);
+            if (defined $expiration_date) {
+              if ($expiration_date eq '') {
+                $expiration_date = undef;
+              } else {
+                $expiration_date_string = $self->_downrez_date($expiration_date);
+              }
             }
             my $creation_date_string = $self->_downrez_date($creation_date);
 
@@ -480,10 +482,12 @@
             ) = @fields[ 1, 3 .. 6, 9, 10 ];
 
             my $expiration_date_string;
-            if ($expiration_date eq '') {
-              $expiration_date = undef;
-            } else {
-              $expiration_date_string = $self->_downrez_date($expiration_date);
+            if (defined $expiration_date) {
+              if ($expiration_date eq '') {
+                $expiration_date = undef;
+              } else {
+                $expiration_date_string = $self->_downrez_date($expiration_date);
+              }
             }
             my $signature_date_string = $self->_downrez_date($signature_date);
 
@@ -552,10 +556,12 @@
             ) = @fields[ 1 .. 11 ];
 
             my $expiration_date_string;
-            if ($expiration_date eq '') {
-              $expiration_date = undef;
-            } else {
-              $expiration_date_string = $self->_downrez_date($expiration_date);
+            if (defined $expiration_date) {
+                if ($expiration_date eq '') {
+                    $expiration_date = undef;
+                } else {
+                    $expiration_date_string = $self->_downrez_date($expiration_date);
+                }
             }
             my $creation_date_string = $self->_downrez_date($creation_date);
 
